server/game/gameSystems/DelayedEffectSystem.ts

```diff
@@ -0,0 +1,93 @@
+import AbilityHelper from '../AbilityHelper';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { IAbilityLimit } from '../core/ability/AbilityLimit';
+import { TriggeredAbilityContext } from '../core/ability/TriggeredAbilityContext';
+import { Duration, EventName } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { GameSystem, IGameSystemProperties } from '../core/gameSystem/GameSystem';
+import { WhenType } from '../Interfaces';
+import * as Contract from '../core/utils/Contract';
+
+export enum DelayedEffectType {
+    Card,
+    Player
+}
+
+export interface IDelayedEffectSystemProperties extends IGameSystemProperties {
+    title: string;
+    when: WhenType;
+    duration?: Duration;
+    limit?: IAbilityLimit;
+    immediateEffect: GameSystem<TriggeredAbilityContext>;
+    effectType: DelayedEffectType;
+}
+
+export class DelayedEffectSystem<TContext extends AbilityContext = AbilityContext> extends GameSystem<TContext, IDelayedEffectSystemProperties> {
+    public override readonly name: string = 'applyDelayedEffect';
+    public override readonly eventName: EventName = EventName.OnEffectApplied;
+    public override readonly effectDescription: string = 'apply a delayed effect';
+
+    protected override defaultProperties: IDelayedEffectSystemProperties = {
+        title: '',
+        when: null,
+        duration: Duration.Persistent,
+        limit: AbilityHelper.limit.perGame(1),
```

Comment:
We should avoid importing AbilityHelper outside of card implementations, it imports a huge amount of other files so it really increases the likelihood of import loops. We can directly import `PerGameAbilityLimit` and `OngoingEffectLibrary`

Reply:
I think I need to add an export on PerGameAbilityLimit? It isn't currently referenceable I think

Reply:
Oh sorry, you can just import `perGame()` looks like